kiagnose/internal/checkup/namespace/namespace.go

```diff
@@ -44,30 +44,36 @@ func Create(client kubernetes.Interface, namespace *corev1.Namespace) (*corev1.N
 
 // DeleteAndWait delete and waits for the given namespace to dispose.
 func DeleteAndWait(client kubernetes.Interface, nsName string, timeout time.Duration) error {
-	if err := client.CoreV1().Namespaces().Delete(context.Background(), nsName, metav1.DeleteOptions{}); err != nil {
-		return err
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	nsLabel := fmt.Sprintf("%s=%s", corev1.LabelMetadataName, nsName)
+	nsWatcher, err := client.CoreV1().Namespaces().Watch(ctx, metav1.ListOptions{LabelSelector: nsLabel})
```

Comment:
There is a "labels" package at apimachinery with some helpers for this operation, what we can use here is "Set" We can use https://pkg.go.dev/k8s.io/apimachinery@v0.24.0/pkg/labels#Set so we don't have to create a expression.

```suggestion
	nsLabel := labels.Set(map[string]string{corev1.LabelMetadataName: nsName})
	nsWatcher, err := client.CoreV1().Namespaces().Watch(ctx, metav1.ListOptions{LabelSelector: nsLabel.String()})
```

Reply:
It seems an overkill to me in this case. We are not using any interesting helper method on the `Set`, I think the current simpler (less dependencies) is good enough.

Reply:
The project is already using apimachinery so the dependency is already there, also I think we can simplify this with 

```
labels.Set{corev1.LabelMetadataName: nsName}
```

Reply:
Done